### PR TITLE
GHA/windows: do not disable TrackFileAccess

### PR DESCRIPTION
**Pull request by AkiSakurai**
To support incremental build, it is necessary to track file access patterns.

### CODE REVIEW

**Comment by vszakats** [comment]
When would the CI do an incremental build?

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -623,7 +622,7 @@ jobs:
             TFLAGS+=' ~614'  # 'SFTP pre-quote chmod' SFTP, pre-quote, directory
           fi
           PATH="$PWD/bld/lib:$PATH:/c/Program Files (x86)/stunnel/bin:/c/Program Files/OpenSSH-Win64"
-          cmake --build bld --config '${{ matrix.type }}' --target test-ci
+          cmake --build bld --config '${{ matrix.type }}' --parallel 5 --target test-ci
```
Comment (vszakats) [inline .github/workflows/windows.yml:625]: This command doesn't actually do a build, but instead launches a single runtests command to run the test (wherr parallelism is controlled by tflags -j option). Or am I missing something?

**Comment by AkiSakurai** [comment]
> When would the CI do an incremental build?

The test-ci target depends on the tests and libcurl targets. If there is no incremental build, executing test-ci will rebuild what was already done in previous steps.

**Comment by AkiSakurai** [comment]
The single-binary test bundle is so fast that it doesn't matter now.

**Comment by vszakats** [comment]
> > When would the CI do an incremental build?
> 
> The test-ci target depends on the tests and libcurl targets. If there is no incremental build, executing test-ci will rebuild what was already done in previous steps.

That's interesting! It explains those extra invocations in `test-ci`.
Also very odd that MSBuild needs extra (and slow) "file access tracking" 
to compare the timestamps of build step outputs/inputs?

According to [docs](https://learn.microsoft.com/en-us/dotnet/api/microsoft.build.tasks.generateresource.trackfileaccess?view=msbuild-17-netcore) this option should only affect incremental
builds. But there is no incremental build here: The inputs (including their
timestamps) do not change between executing these steps. Yet, it
looks like MSBuild is doing something weird and/or thinks they did, 
considers this incremental, and builds again, while there shouldn't be a
new build necessary in the first place?

Since file tracking makes builds noticeably slower (for non-unity
builds in particular), would it perhaps be a better way to tackle this
by dropping `DEPENDS testdeps` from the `test-ci` (and friends)
targets?

This would also make GNU Make jobs skip the phase of stepping
through all tests and doing a no-op build to build them again. (In 
those cases there doesn't seem to be an actual rebuild, but it's still
taking significant time on Windows.) Interestingly, ninja skips that
the idle work automatically.

ninja: https://github.com/curl/curl/actions/runs/10975877072/job/30483587344#step:25:1
GNU Make (old mingw-w64): https://github.com/curl/curl/actions/runs/10975877072/job/30483586469#step:14:11

`test-ci` never worked to build everything necessary to run the tests,
notably the curl tool is missed. (Same for autotools, where libcurl and
curl tool are both missed.)

Or omit `testdeps` with an option. Or just run `runtests.pl` manually
in CI.

(You may be interested in the #14772 that I just merged, which should
make all this much less costly, because libtests and unit tests are built
as a single binary each.)

**Comment by vszakats** [comment]
Opened this: #15001

**Pull request closed without merging**